app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1186,14 +1193,39 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope {
             }
         }
 
+        private fun animateProgressBar(newProgress: Int, isLoading: Boolean) {
+            progressBarAnimation?.cancel()
+
+            if (pageLoadingIndicator.progress > newProgress) {
+                pageLoadingIndicator.progress = 0
+            }
+
+            progressBarAnimation = ObjectAnimator.ofInt(pageLoadingIndicator, "progress", newProgress).apply {
```

Comment:
could this `ObjectAnimator` be moved into an individual class?

My motivation is just to avoid this class to get bigger, and avoid adding distraction when adding new features, if moving it away keeps the same size or makes the read more complex we can keep it here.

Reply:
I can't see how to do that without making it more difficult to understand. `ObjectAnimator` is a class in itself and here we are creating one with a listener. What I can do is using the crossinline listeners to remove noise though.